.github/workflows/documentation.yaml

```diff
@@ -0,0 +1,20 @@
+name: documentation
+on:
+  push:
+    branches:
+      - master
+      - main
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.x
+      - name: Install mkdocs requirements
+        run: pip install -r requirements.txt
+        working-directory: docs
+      - name: Publish documentation to gp-pages branch
+        run: mkdocs gh-deploy --force
```

Comment:
This will deploy into `gh-pages` branch and that is then deployed by Github Pages